src/plugins/banner/public/plugin.ts

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from '../../../core/public';
+import { BannerPluginSetup, BannerPluginStart } from './types';
+import { bannerServiceInstance } from './services/banner_service_instance';
+import { GlobalBanner } from './components/global_banner';
+
+export class BannerPlugin implements Plugin<BannerPluginSetup, BannerPluginStart> {
+  constructor(_initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup): BannerPluginSetup {
+    // Setup banner with default values
+    bannerServiceInstance.setup({
```

Comment:
default values from come from yml. Are you planning to implement this in the future, if that's the case, can we add a TODO here in the code